packages/query-core/src/types.ts

```diff
@@ -27,7 +27,15 @@ export interface QueryFunctionContext<
 
 export type InitialDataFunction<T> = () => T | undefined
 
-export type PlaceholderDataFunction<TResult> = () => TResult | undefined
+type NonFunctionGuard<T> = T extends Function ? never : T
+
+export type PlaceholderDataFunction<TQueryData> = (
+  previousData: TQueryData | undefined,
+) => TQueryData | undefined
+
+export type QueriesPlaceholderDataFunction<TQueryData> = () =>
+  | TQueryData
+  | undefined
```

Comment:
let's also document please that `useQueries` doesn't get previousData passed because the feature doesn't really work reliably with the dynamic nature of useQueries

Reply:
✅